### PR TITLE
Implement batch file uploads

### LINKED ISSUE #672: Allow batch file uploads on admin panel

### PR DESCRIPTION
**Pull request by Isti01**
closes #672

### CODE REVIEW

File: backend/src/main/kotlin/hu/bme/sch/cmsch/controller/admin/FileUploadController.kt [hunk 1]
```diff
@@ -69,13 +71,19 @@ class FileUploadController(
...
-        val newName = name.replace(" ", "_").replace(Regex("[^A-Za-z0-9_]+"), "").uppercase() +
-                "_${Random().nextLong().absoluteValue.toString(36).uppercase()}" +
-                originalFilename.substring(if (originalFilename.contains(".")) originalFilename.lastIndexOf('.') else 0)
-        file?.uploadFile("public", newName)
-
-        return "redirect:/admin/control/upload-file?uploaded=${newName}"
+        if (names.size != files.size) {
+            throw IllegalArgumentException("The length of names and files does not match")
+        }
+        val newNames = ArrayList<String>()
```
Comment (Gerviba) [inline backend/src/main/kotlin/hu/bme/sch/cmsch/controller/admin/FileUploadController.kt:77]: I'd use arrayListOf() vagy mutableListOf()

File: backend/src/main/resources/templates/uploadPhoto.html [hunk 2]
```diff
@@ -120,6 +121,15 @@ <h5 th:text="${permission}">PERMISSION_NAME</h5>
...
 </script>
-
+<template id="name-input-template">
+    <div class="field-group">
+        <label class="name-label">Cím a galériában
+            <span class="file-name"></span>
+        </label>
+        <div style="clear: both"></div>
+        <input type="text" class="name-input" name="names" autocomplete="off" />
+        <small>Vessző ne legyen a névben</small>
```
Comment (Gerviba) [inline backend/src/main/resources/templates/uploadPhoto.html:131]: I'd just remove them from the name if there is any. Maybe mention that special chars will be removed.

**Review by Gerviba — APPROVED** [review]
Good one!

File: backend/src/main/resources/templates/uploadPhoto.html (hunk 2 above)
Comment (Isti01) [reply backend/src/main/resources/templates/uploadPhoto.html:131]: I take the file names as a list of strings and spring fails to parse it if there are commas inside

**Pull request merged**